engines/suppliers/app/controllers/suppliers/api/v1/suppliers_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Suppliers
+  module Api
+    module V1
+      class SuppliersController < ApplicationController
+        before_action :load_supplier, only: [:show, :update, :destroy]
+
+        # GET /api/v1/suppliers
+        #
+        def index
+          suppliers = SuppliersFinder.new(supplier_params[:group_id]).find
```

Comment:
btw, do you have any working example? I haven't used repositories yet and I'm still figuring how all the pieces fit together.